brightness-mode-explainer.md

```diff
@@ -0,0 +1,128 @@
+  - Idea is to focus on mobile first (i.e. not worry about external displays). How to indicate that in the spec? Does it *need* to be indicated in the spec?
+  - Should this trigger "`prefers-contrast: more`" in [CSS Media Queries](https://drafts.csswg.org/mediaqueries-5/#prefers-contrast)?
+- From [https://github.com/w3c/devicesensors-wg/issues/51](https://github.com/w3c/devicesensors-wg/issues/51)
+  - Brightness levels
+    - How bright is too bright? Consider 100% brightness with HDR displays, for example.
+    - Take a granular value or an enum?
+    - Related to whether script should be allowed to reduce brightness.
+  - Permission model
+    - Probably require user gesture (request but not consume it).
+    - How to deal with multiple or external displays
+  - While brightness changes
+    - What if users change the brightness level while the lock is held?
+  - When dropping screen brightness request
+      - Does the UA have to restore the previous brightness level?
+        - What about external displays? Do UAs need to keep track of levels for each display?
+      - Should script be allowed to "hold the lock" indefinitely?
+
+## Goals
+
+- Provide the ability to request an increase to the display's brightness. This could be a UA-controlled maximum (which *could* be the maximum display brightness), either indefinitely or for a period of time.
+- Provide the ability to release the request so that the device's brightness returns to its pre-request value (i.e. hand back control to OS).
+- Handle error cases, where such requests are denied or not possible.
+
+## Non-goals
+
+- Reading screen brightness level.
+- Adjusting display brightness level to arbitrary values (absolute or relative).
+  - [\<video\> integration to allow e.g. granular brightness control](https://github.com/w3c/screen-wake-lock/issues/129#issuecomment-926603108) is an instance of the above. This is most likely better handled elsewhere and should probably be a UA-specific control.
+
+## Proposed Solutions
+
+This was initially discussed in [Screen Wake Lock API issue 129](https://github.com/w3c/screen-wake-lock/issues/129) back in 2018 and also proposed as a [separate API in WICG](https://github.com/WICG/proposals/issues/17). This topic was also discussed at multiple TPACs under the Devices and Sensors WG. As such, several different API shapes have been considered over the years.
+
+At this point, none have been fully discarded, and we are open to feedback about adopting some of the ideas described here.
+
+### Screen IDL interface extension
+
+Extend the [Screen interface](https://drafts.csswg.org/cssom-view/#the-screen-interface) with a new operation, inspired by the [Wake Lock API](https://developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel). Something like:
+
+```webidl
+[SecureContext]
+partial interface Screen {
+  Promise<ScreenBrightnessSentinel> requestBrightnessIncrease();
+};
+
+interface ScreenBrightnessSentinel {
+  Promise<undefined> release();
```

Comment:
Nit: I'm not sure returning a promise is required.

Reply:
It's useful to do this and have the promise fulfil when the release completes. This would be useful if vendors implement brightness changes as transitions rather than a binary toggle.

Entering and exiting fullscreen is asynchronous so this would work in a similar way.

Reply:
See Android for instance: https://developer.android.com/reference/android/view/WindowManager.LayoutParams#screenBrightness

Reply:
@beaufortfrancois You're 100% correct. I don't know whether this is going to change in the future and therefore if it's worth having the vendor return a fulfilled promise in case it does?

Reply:
From experience with screen orientation lock, I'd also be inclined to make it async. 

Although changing the brightness is sync at the OS level, there might be some IPC involved and stuff can get out of sync. Worst case, we just return `Promise.resolve()`.

Reply:
Let's keep this as-is for now; there is still time to change it before anyone implements any of the ideas here, and using a Promise is more future-proof.